sections/git_branch.zsh

```diff
@@ -25,6 +37,25 @@ spaceship_git_branch() {
   git_current_branch="${git_current_branch#heads/}"
   git_current_branch="${git_current_branch/.../}"
 
+  # Build commit SHA info if enabled
+  if [[ $SPACESHIP_GIT_BRANCH_SHOW_COMMIT == true ]]; then
+    local git_commit_prefix="$SPACESHIP_GIT_BRANCH_COMMIT_PREFIX$SPACESHIP_GIT_BRANCH_COMMIT_SYMBOL"
+    local git_commit_sha=$(git rev-parse --short HEAD 2> /dev/null)
+    local git_commit_info="$git_commit_prefix$git_commit_sha$SPACESHIP_GIT_BRANCH_COMMIT_SUFFIX"
+  fi
+
+  git_current_branch+="$git_commit_info"
+
+  # Build tag info if enabled
+  if [[ $SPACESHIP_GIT_BRANCH_SHOW_TAG == true ]]; then
+    local git_tag_prefix="$SPACESHIP_GIT_BRANCH_TAG_PREFIX$SPACESHIP_GIT_BRANCH_TAG_SYMBOL"
+    local git_tag_name=$(git describe --tags --exact-match $git_commit_sha 2>/dev/null)
+    local git_tag_info="$git_tag_prefix$git_tag_name$SPACESHIP_GIT_BRANCH_TAG_SUFFIX"
+    if [[ -n $git_tag_name ]]; then
+      git_current_branch+="$git_tag_info"
+    fi
+  fi
```

Comment:
You could polish this piece of work to look like the commit work: the `git_current_branch+="$git_tag_info"` could be outside and after the entire `[[ $SPACESHIP_GIT_BRANCH_SHOW_TAG == true ]]` conditional. Because like with the commit thing, `x=""; y="hi"; if [[ -n $x ]]; then y+="$x"; fi` has the same result as `x="'; y="hi"; y+="$x"`

Reply:
Mmmm... Not really sure about that, @olets...

What happens if `git describe --tags --exact-match $git_commit_sha 2>/dev/null` returns an empty string? Should the `SPACESHIP_GIT_BRANCH_TAG_SYMBOL` be shown without any kind of relevant information?

![image](https://user-images.githubusercontent.com/879123/40547574-995d0c80-6032-11e8-91a0-1976fe47f9ee.png)

Reply:
Oh yep you're absolutely right! In that case, would it work to put the `git_tag_prefix` and `git_tag_info` inside the `-n $git_tag_name` conditional? Don't need to do that work if there's no tag

Again, I'm not an official Spaceship dev. I'm enjoying this, but you can ignore me if you want :)

Reply:
You're enjoying this, but I'm learning and enjoying too with your guidance... xD 

Than you so much, @olets!!!